platform/mellanox/mlnx-platform-api/sonic_platform/component.py

```diff
@@ -23,6 +24,26 @@
 CPLD_VERSION_FILE_PATTERN = '/var/run/hw-management/system/cpld[0-9]_version'
 CPLD_VERSION_MAX_LENGTH = 4
 
+CPLD_UPDATE_COMMAND = "cpldupdate --dev {} {}"
+CPLD_INSTALL_SUCCESS_FLAG = "PASS!"
+
+MST_DEVICE_PATTERN = "/dev/mst/mt[0-9]*_pciconf0"
+MACHINE_CONF_FILE = "/host/machine.conf"
+MACHINE_CONF_MAX_SIZE = 2048
+
+BIOS_VERSION_PARSE_PATTERN = 'OEM[\s]*Strings\n[\s]*String[\s]*1:[\s]*([0-9a-zA-Z_\.]*)'
+ONIE_VERSION_PARSE_PATTERN = 'onie_version=[0-9]{4}\.[0-9]{2}-([0-9]+)\.([0-9]+)\.([0-9]+)'
+ONIE_VERSION_MAJOR_OFFSET = 1
+ONIE_VERSION_MINOR_OFFSET = 2
+ONIE_VERSION_RELEASE_OFFSET = 3
+# To update BIOS requires the ONIE with version 5.2.0016 or upper
+ONIE_REQUIRED_MAJOR = "5"
+ONIE_REQUIRED_MINOR = "2"
+ONIE_REQUIRED_RELEASE = "0016"
+
+ONIE_FW_UPDATE_CMD_ADD = "/usr/bin/onie-fw-update add {}"
+ONIE_FW_UPDATE_CMD_UPDATE = "/usr/bin/onie-fw-update update"
```

Comment:
Having a separate class looks like overdesign to me. but your opinion regarding the namespace pollute is absolutely correct.
I decide to move these defines to the componentBIOS class and have a function for it.

Reply:
I found that so far the onie version check is only for BIOS because only to update BIOS has a version restriction.
So I decide not to move to the Component class but keep it in the ComponentBIOS class as a method.
Let me know if you have any concern.

Reply:
@stephenxs ok.